release/signing_images.sh

```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
+# Copyright (C) 2018 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# shellcheck source=common.sh
+# shellcheck disable=2162
+# shellcheck disable=2086
+
+set -e
+
+SCRIPT_DIR=$(dirname "$(realpath "${BASH_SOURCE[0]}")")
+
+. "${SCRIPT_DIR}/../common.sh"
+. "${SCRIPT_DIR}/../config/config.sh"
+
+var_load MIX_VERSION
+
+IMGS_DIR="${WORK_DIR}/release/images"
+LOG_DIR="${WORK_DIR}/logs"
+PROCS_PER_IMG=8
+TEMPLATES_PATH=${TEMPLATES_PATH:-"${PWD}/config/images"}
+SIGN_FILE_SUFFIX=${SIGN_FILE_SUFFIX:-"SHA512SUM"}
+SHASUM_CMD=${SHASUM_CMD:-"sha512sum"}
+IMAGE_EXT=${IMAGE_EXT:-"img"}
+COMPRESSION_EXT=${COMPRESSION_EXT:-"xz"}
+
+signing_images() {
+    local image=$1
+    local name=$(basename "${image%.json}")
+
+    # image is needed to exists before shasum and signing
+    local image_file="${DISTRO_NAME}-${MIX_VERSION}-${name}.${IMAGE_EXT}.${COMPRESSION_EXT}"
+    # set the shasum and signature name
+    local shasum_file="${image_file}-${SIGN_FILE_SUFFIX}"
+    local signature_file="${shasum_file}.sig"
+
+    # going into the directory avoids inclusion of folder during shasum generation
+    pushd ${IMGS_DIR}
+
+    # make sure the images are already generated
+    if [[ ! -f "${IMGS_DIR}/${image_file}" ]]; then
+        log "image file ${image_file} does not exist. Please generate the images prior to signing."
+        return 1
+    fi
+
+    # shasum image
+    log "Generating ${IMGS_DIR}/${shasum_file} ..."
+    eval "${SHASUM_CMD} ${image_file} > ${shasum_file}"
+    # default signing with openssl using sha1 to make it work out of the box
+    if [[ -z "${CUSTOM_SIGN_CMD}" ]]; then
+        log "Generating ${IMGS_DIR}/${signature_file} ..."
+        eval "openssl dgst -sha1 -sign ${PRIVATE_KEY} -out ${signature_file} ${shasum_file}"
+    else
+        # customized signage command
+        log "signing with custom command"
+        eval "${CUSTOM_SIGN_CMD}"
```

Comment:
Even that is a custom cmd, I would expect to recebe at the the file as input. Otherwise you would have to assume variable names when exporting the custom command.

Reply:
do you want me to change this part?